src/environment_setup/python_package.ml

```diff
@@ -27,7 +27,10 @@ let create_python_tool ~host ~(run_program : Machine.Make_fun.t) ~install_path
     | (Conda, Package_PyPI pname) -> ["conda"; "skeleton"; "pypi"; pname], pname
     | _ -> failwith "Installation type not supported."
   in
-  let conda_env = Conda.setup_environment ~python_version install_path name in
+  let main_subdir = name ^ "_conda_dir" in
+  let conda_env = 
+    Conda.setup_environment ~python_version ~main_subdir install_path name
+  in
```

Comment:
Yes That maybe be better; because even with the lock I get weird errors like

```
Traceback (most recent call last):
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/cli/main.py", line 139, in main
    args_func(args, p)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/cli/main.py", line 146, in args_func
    args.func(args, p)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/cli/main_create.py", line 49, in execute
    install.install(args, parser, 'create')
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/cli/install.py", line 424, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/plan.py", line 539, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/instructions.py", line 149, in execute_instructions
    cmd(state, arg)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/instructions.py", line 92, in LINK_CMD
    link(state['prefix'], arg, index=state['index'])
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/instructions.py", line 88, in link
    install.link(pkgs_dir, prefix, dist, lt, index=index)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/install.py", line 660, in link
    update_prefix(join(prefix, f), prefix, placeholder, mode)
  File "/nfswork/biokepi-work-dir/toolkit/python-tools/conda_dir/lib/python3.5/site-packages/conda/install.py", line 277, in update_prefix
    with open(path, 'rb') as fi:
FileNotFoundError: [Errno 2] No such file or directory: '/nfswork/biokepi-work-dir/toolkit/python-tools/envs/topiary/bin/c_rehash'
```